api/app/src/providers/withings.ts

```diff
@@ -113,6 +113,13 @@ export class Withings extends Provider implements OAuth2 {
           }
         );
 
+        const statusOk = 0;
+
+        if (response.data.status !== statusOk) {
+          console.log(response.data);
+          throw new Error(response.data.error);
+        }
+
```

Comment:
So i couldn't get down to why initially the token was bad, but if there is an error from withings it returns and error object so this is to handle that and not override the providermap with bad data